### PR TITLE
biosig: fix Sonoma build

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/b/biosig.rb [hunk 1]
```diff
@@ -26,12 +26,19 @@ class Biosig < Formula
   depends_on "suite-sparse"
   depends_on "tinyxml2"
 
+  on_sonoma :or_newer do
+    depends_on "libiconv"
```
Comment (Bo98) [inline Formula/b/biosig.rb:30]: Can you explain the reason for this? A lot of `iconv` issues have been fixed so would be good to know what issues are remaining.

Comment (cho-m) [reply Formula/b/biosig.rb:30]: Main issue looks like code doesn't handle BSD iconv correctly https://git.ista.ac.at/alois.schloegl/code/-/blob/master/biosig4c++/t210/sopen_scp_read.c?ref_type=heads#L368

```c
#if  defined(_ICONV_H) || defined (_LIBICONV_H)
```

At minimum `_ICONV_H_` is guard. Maybe just modifying that could work. Not sure which header `_ICONV_H` would be from.

Comment (Bo98) [reply Formula/b/biosig.rb:30]: `_ICONV_H` is glibc iconv.
`_LIBICONV_H` is GNU libiconv
`_ICONV_H_` is BSD iconv.

Though those are internal defines so it probably shouldn't really be checking for things like that and should be doing configure checks instead. But as a quick patch then adding `_ICONV_H_` should work.

**Review by carlocab — APPROVED** [review]

**Review by Bo98 — APPROVED** [review]
Looks good but needs an upstream report

**Review by carlocab — APPROVED** [review]

**Pull request merged**